src/sensors/can/can_manager.hpp

```diff
@@ -0,0 +1,108 @@
+class CANManager {
+public:
+    /// @brief Default constructor. FlexCAN moment for forcing the definition to be in the cpp
+    CANManager();
+
+    /// @brief Deleted copy and assignment constructors. This class should only ever have one instance
+    /// @param copy The CANManager object to copy
+    CANManager(const CANManager& copy) = delete;
+    /// @brief Deleted copy and assignment constructors. This class should only ever have one instance
+    /// @param copy The CANManager object to copy
+    /// @return CANManager& The reference to this object
+    CANManager& operator=(const CANManager& copy) = delete;
+
+    /// @brief Destructor, cleans up motor array
+    ~CANManager();
+
+public:
+    /// @brief Initialize the CAN buses and motor array
+    /// @note This can be called multiple times in the event of hot reloading
+    void init();
+
+    /// @brief Dynamically create the motor objects based on config data
+    /// @param motor_info Motor info array from the config yaml. 2D array holding information in the form: CAN_MAX_MOTORS * [motor_controller_type, per_bus_motor_id, bus_id]
+    void configure(float motor_info[CAN_MAX_MOTORS][3]);
+
+    /// @brief Read data from all busses and distribute them to the correct motors
+    void read();
+
+    /// @brief Issue write commands to all motors on each bus
+    /// @note This issues a CAN command over the bus
+    void write();
+
+    /// @brief Issue zero torque commands to all motors
+    /// @note This immediately issues a CAN command over the bus
+    void safety_mode();
+
+    /// @brief Write a torque command to a specific motor given it's global ID
+    /// @param motor_gid The global ID of the motor to write to
+    /// @param torque The normalized torque value to write to the motor in the range [-1, 1]
+    /// @note This does not issue a CAN command over the bus
+    void write_motor_torque(uint32_t motor_gid, float torque);
+
+    /// @brief Print the state of all motors
+    void print_state();
+
+    /// @brief Print the state of a specific motor
+    /// @param motor_gid The global ID of the motor to print the state of
+    void print_motor_state(uint32_t motor_gid);
+
+    /// @brief Get the underlying motor object by ID
+    /// @param motor_gid The global ID of the motor to get
+    /// @return The motor object if it exists, nullptr if it does not
+    /// @note You can use the motor's get_type to determine the type of motor and dynamic_cast to the correct motor type
+    Motor* get_motor(uint32_t motor_gid);
+
+private:
+    /// @brief Verify that all motors are online and ready
+    void init_motors();
+
+    /// @brief Iterates through all the motors and tries to give the message to the correct one
+    /// @param msg The message to distribute
+    /// @return True if the message was distributed, false if it was not
+    bool distribute_msg(CAN_message_t& msg);
+
+private:
+    /// @brief CAN bus 1
+    FlexCAN_T4<CAN1, RX_SIZE_256, TX_SIZE_16> m_can1;
```

Comment:
Can you add a comment here about things breaking if this is constructed again? When this happens, why this happens (even if that's just flexcan weird library issues)

Also consider static keyword